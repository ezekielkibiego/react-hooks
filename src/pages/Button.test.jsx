/* eslint-disable no-undef */
import { fireEvent, render, screen } from "@testing-library/react"
import Button from "./Button"
import React from "react";

test("renders button with correct label", () => {
    render(<Button label="Click me" />);
    const buttonElement = screen.getByText(/Click me/i);
    expect(buttonElement).toBeInTheDocument();
});

test("calls onClick when clicked", () => {
    const handleClick = jest.fn();
    render(<Button label="Click" onClick={handleClick} />);
    const buttonElement = screen.getByText(/Click/i);
    fireEvent.click(buttonElement);
    expect(handleClick).toHaveBeenCalledTimes(1);
});

test("Button matches snapshot", () => {
    const { asFragment } = render(<Button label="Click me" />);
    expect(asFragment()).toMatchSnapshot();
});