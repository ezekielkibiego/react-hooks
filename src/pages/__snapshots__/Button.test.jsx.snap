// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <button>
      Click me
    </button>
  </div>
</DocumentFragment>
`;
